components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/WorkflowsApiServiceImpl.java

```diff
@@ -0,0 +1,207 @@
+        ApiMgtDAO apiMgtDAO = ApiMgtDAO.getInstance();
+        boolean isTenantFlowStarted = false;
+        String username = RestApiCommonUtil.getLoggedInUsername();
+        String tenantDomainOfUser = MultitenantUtils.getTenantDomain(username);
+        try {
+            if (workflowReferenceId == null) {
+                RestApiUtil.handleBadRequest("workflowReferenceId is empty", log);
+            }
+            org.wso2.carbon.apimgt.impl.dto.WorkflowDTO workflowDTO = apiMgtDAO.retrieveWorkflow(workflowReferenceId);
+            if (workflowDTO == null) {
+                RestApiUtil.handleResourceNotFoundError(RestApiConstants.RESOURCE_WORKFLOW, workflowReferenceId, log);
+            }
+            String tenantDomain = workflowDTO.getTenantDomain();
+            if (tenantDomain != null && !tenantDomain.equals(tenantDomainOfUser)) {
+                return Response.status(Response.Status.UNAUTHORIZED).build();
+            }
+            if (tenantDomain != null && !SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {
+                isTenantFlowStarted = true;
+                PrivilegedCarbonContext.startTenantFlow();
+                PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain, true);
+            }
+            if (body == null) {
+                RestApiUtil.handleBadRequest("Request payload is missing", log);
+            }
+            if (body.getDescription() != null) {
+                workflowDTO.setWorkflowDescription(body.getDescription());
+            }
+            if (body.getStatus() == null) {
+                RestApiUtil.handleBadRequest("Workflow status is not defined", log);
+            } else {
+                workflowDTO.setStatus(WorkflowStatus.valueOf(body.getStatus().toString()));
+            }
+            if (body.getAttributes() != null) {
+                workflowDTO.setAttributes(body.getAttributes());
+            }
+            String workflowType = workflowDTO.getWorkflowType();
+            if(WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_CREATION.equals(workflowType)) {
+                WorkflowExecutor workflowExecutor = WorkflowExecutorFactory.getInstance().getWorkflowExecutor(workflowType);
+                workflowExecutor.complete(workflowDTO);
+                if (WorkflowStatus.APPROVED.equals(workflowDTO.getStatus())) {
+                    WorkflowUtils.sendNotificationAfterWFComplete(workflowDTO, workflowType);
+                }
+                APIUtil.logAuditMessage(APIConstants.AuditLogConstants.WORKFLOW_STATUS, new Gson().toJson(workflowDTO),
+                        APIConstants.AuditLogConstants.UPDATED, RestApiCommonUtil.getLoggedInUsername());
+                return Response.ok().entity(body).build();
+            } else {
+                RestApiUtil.handleBadRequest("Invalid Workflow Type", log);
+            }
+        } catch (APIManagementException e) {
+            String msg = "Error while resuming workflow " + workflowReferenceId;
```

Comment:
No need of assigning a new `msg` variable.

Reply:
Refactored